onecodex/viz/_bargraph.py

```diff
@@ -134,82 +154,65 @@ def plot_bargraph(
         # OCX this should be okay)
         #
 
-        ylabel = df.ocx.field if ylabel is None else ylabel
+        ylabel = field if ylabel is None else ylabel
         xlabel = "" if xlabel is None else xlabel
 
         # should ultimately be Label, tax_name, readcount_w_children, then custom fields
         tooltip_for_altair = [magic_fields[f] for f in tooltip]
         tooltip_for_altair.insert(1, "tax_name")
-        tooltip_for_altair.insert(2, "{}:Q".format(df.ocx.field))
+        tooltip_for_altair.insert(2, "{}:Q".format(field))
 
-        # generate dataframes to plot, one per facet
-        dfs_to_plot = []
+        kwargs = {}
 
         if haxis:
-            # if using facets, first facet is just the vertical axis
-            blank_df = df.iloc[:1].copy()
-            blank_df[df.ocx.field] = 0
-
-            dfs_to_plot.append(blank_df)
-
-            for md_val in magic_metadata[magic_fields[haxis]].unique():
-                # special case where the metadata value is None: must use isnull()
-                if md_val is None:
-                    plot_df = df.where(df[magic_fields[haxis]].isnull()).dropna(how="all")
-                else:
-                    plot_df = df.where(df[magic_fields[haxis]] == md_val).dropna(how="all")
-
-                # preserve booleans
-                if magic_metadata[magic_fields[haxis]].dtype == "bool":
-                    plot_df[magic_fields[haxis]] = plot_df[magic_fields[haxis]].astype(bool)
-
-                dfs_to_plot.append(plot_df)
-        else:
-            dfs_to_plot.append(df)
-
-        charts = []
-
-        for plot_num, plot_df in enumerate(dfs_to_plot):
-            if sort_x:
-                sort_order = sort_x(plot_df["Label"].tolist())
-
-            chart = (
-                alt.Chart(plot_df)
-                .mark_bar()
-                .encode(
-                    x=alt.X("Label", axis=alt.Axis(title=xlabel), sort=sort_order),
-                    y=alt.Y(
-                        df.ocx.field,
-                        axis=alt.Axis(title=ylabel),
-                        scale=alt.Scale(domain=[0, 1], zero=True, nice=False),
-                    ),
-                    color=alt.Color("tax_name", legend=alt.Legend(title=legend)),
-                    tooltip=tooltip_for_altair,
-                    href="url:N",
-                )
+            kwargs["column"] = haxis
+
+        domain = sorted(df["tax_name"].unique())
+
+        no_level_name = "No {}".format(rank)
```

Comment:
I think we can generate all of those at the module level in a dict? Not that this would make any differences for performance but maybe a bit cleaner.